src/lib.rs

```diff
@@ -672,16 +673,25 @@ impl ConfigFile {
     }
   }
 
-  pub fn to_import_map_path(&self) -> Option<String> {
-    self.json.import_map.clone()
-  }
-
-  pub fn node_modules_dir_flag(&self) -> Option<bool> {
-    self.json.node_modules_dir
-  }
-
-  pub fn vendor_dir_flag(&self) -> Option<bool> {
-    self.json.vendor
+  pub fn to_import_map_specifier(&self) -> Result<Option<Url>, AnyError> {
+    let Some(value) = self.json.import_map.as_ref() else {
+      return Ok(None);
+    };
+    // try to resolve as a url
+    if let Ok(specifier) = Url::parse(value) {
+      return Ok(Some(specifier));
+    }
+    // now as a relative file path
+    if let Ok(config_file_path) = self.specifier.to_file_path() {
+      let path = config_file_path.parent().unwrap().join(value);
+      Url::from_file_path(&path)
+        .map_err(|()| {
+          anyhow::anyhow!("Failed converting {} to url.", path.display())
+        })
+        .map(Some)
+    } else {
+      Ok(Some(self.specifier.join(value)?))
+    }
   }
```

Comment:
Moved from the CLI: https://github.com/denoland/deno/blob/ae703041b1921affb7fa8a0aa865c6f302c72d6e/cli/args/mod.rs#L1707-L1739